src/Statement/Grouper/FilteringStatementGrouper.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Statement\Grouper;
+
+use InvalidArgumentException;
+use Wikibase\DataModel\Statement\Statement;
+use Wikibase\DataModel\Statement\StatementFilter;
+use Wikibase\DataModel\Statement\StatementList;
+
+/**
+ * @since 3.2
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class FilteringStatementGrouper implements StatementGrouper {
```

Comment:
See my comment in https://gerrit.wikimedia.org/r/#/c/254394/3/view/src/ItemView.php.